includes/acl/core/error_exceptions.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+////////////////////////////////////////////////////////////////////////////////
+// The MIT License (MIT)
+//
+// Copyright (c) 2017 Nicholas Frechette & Animation Compression Library contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+////////////////////////////////////////////////////////////////////////////////
+
+#include <cstdio>
+#include <cstdarg>
+#include <stdexcept>
+
+#if !defined(ACL_NO_ERROR_CHECKS)
+	// Always enabled by default
+	#define ACL_USE_ERROR_CHECKS
+#endif
+
+class AssertFailed : public std::exception
```

Comment:
Why are these 2 classes not in the acl namespace? I'm also not sure if these belong under core as opposed to the unit test project; is this reasonably expected to be used by something else? In fact, I was thinking of removing even the shims/stubs from error.h and just have empty asserts/ensure by default and leave it to every game engine and application to override the behavior, same as allocators.

Reply:
Forgot the namespace, will fix.

I'm not planning to use the exceptions for anything but testing. I had a suspicion end users might want the feature, in which case it can be easily enabled with a single #include. Asserts are used for different failure conditions than exceptions though, so this is unlikely.